Doppler.BillingUser/Controllers/WebhooksController.cs

```diff
@@ -0,0 +1,91 @@
+{
+    [ApiController]
+    public class WebhooksController : ControllerBase
+    {
+        private readonly IBillingRepository _billingRepository;
+        private readonly ICurrencyRepository _currencyRepository;
+        private readonly ILogger<WebhooksController> _logger;
+        private readonly IUserRepository _userRepository;
+        private readonly IMercadoPagoService _mercadoPagoService;
+        private readonly IPaymentStatusMapper _paymentStatusMapper;
+        private readonly string PAYMENT_UPDATED = "payment.updated";
+
+        public WebhooksController(
+            ICurrencyRepository currencyRepository,
+            ILogger<WebhooksController> logger,
+            IBillingRepository billingRepository,
+            IUserRepository userRepository,
+            IMercadoPagoService mercadoPagoService,
+            IPaymentStatusMapper paymentStatusMapper)
+        {
+            _billingRepository = billingRepository;
+            _currencyRepository = currencyRepository;
+            _logger = logger;
+            _userRepository = userRepository;
+            _mercadoPagoService = mercadoPagoService;
+            _paymentStatusMapper = paymentStatusMapper;
+        }
+
+        [HttpPost("/accounts/{accountname}/integration/mercadopagonotification")]
+        public async Task<IActionResult> UpdateMercadoPagoPaymentStatusAsync([FromRoute] string accountname, [FromBody] MercadoPagoNotification notification)
+        {
+            if (notification.Action != PAYMENT_UPDATED)
+            {
+                return new OkObjectResult("Successful");
+            }
+
+            var user = await _userRepository.GetUserInformation(accountname);
+            if (user is null)
+            {
+                return new NotFoundObjectResult("Account not found");
+            }
+
+            var invoice = await _billingRepository.GetInvoice(user.IdUser, notification.Data.Id.ToString());
+            if (invoice is null)
+            {
+                _logger.LogError("Invoice with authorization number: {0} was not found.", notification.Data.Id);
+                return new NotFoundObjectResult("Invoice not found");
+            }
+
+            var payment = await _mercadoPagoService.GetPaymentById(notification.Data.Id, accountname);
+
+            var status = _paymentStatusMapper.MapToPaymentStatus(payment.Status);
+            if (status == PaymentStatusEnum.Pending && invoice.Status == PaymentStatusEnum.Approved)
+            {
+                return new OkObjectResult("Successful");
+            }
+
+            if (status == PaymentStatusEnum.DeclinedPaymentTransaction && invoice.Status != PaymentStatusEnum.DeclinedPaymentTransaction)
+            {
+                if (invoice.Status == PaymentStatusEnum.Approved)
+                {
+                    _logger.LogError($"The payment associated to the invoiceId {invoice.IdAccountingEntry} was rejected. Reason: {payment.StatusDetail}");
```

Comment:
I think you should update this message. See https://docs.microsoft.com/en-US/dotnet/fundamentals/code-analysis/quality-rules/ca2254
```suggestion
                    _logger.LogError("The payment associated to the invoiceId {invoiceId} was rejected. Reason: {reason}", invoice.IdAccountingEntry, payment.StatusDetail);
```